src/array.zig

```diff
@@ -1,5 +1,5 @@
 const std = @import("std");
-const parser = @import("./parser.zig");
+const parser = @import("parser");
```

Comment:
Why these changes?

Reply:
Since the serializer lives under the `serialize` folder, we cannot simply import datetime.zig by writing `import("../datetime.zig")` and rather need to declare it as nodule. Also parser.zig is required by datetime.zig and also by other files so we need to declare it as a module too.

Reply:
> Since the serializer lives under the serialize folder, we cannot simply import datetime.zig by writing import("../datetime.zig")

Why not? It perfectly works. You can import any zig files you need. `zig-toml` is a small library, I think it's a bit overkill to start with modules. A nice example is the ziglang code. It has just few of them, one for the compile, another for `std` and a few others.

IIUC you added those modules to be able to access the three date types. I guess you can just
```zig
const datetime = @import("../datetime.zig");
```
like in `main.zig` and a few others.

Reply:
Yeah I ran into the issue because I had declared src/serialize/tests.zig as a separate test step which obviously cannot allow import from upper directory levels. Instead now I just import the that file in the src/tests.zig file which resolves the issue.